src/main/java/org/usfirst/frc4048/common/loggingv2/CommandLogger.java

```diff
@@ -0,0 +1,53 @@
+package org.usfirst.frc4048.common.loggingv2;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.*;
+
+public class CommandLogger {
+    private final LinkedHashMap<Command, Queue<Boolean>> toLogCommandStatus = new LinkedHashMap<>();
+    private static final CommandLogger inst = new CommandLogger();
+    private boolean hasInit;
+
+    public static CommandLogger get() {
+        return inst;
+    }
+
+    private CommandLogger() {}
+
+    public void init() {
+        if (hasInit) return;
+        CommandScheduler.getInstance().onCommandInitialize(command -> {
+            Queue<Boolean> toLogBools = toLogCommandStatus.getOrDefault(command, new LinkedList<>());
+            toLogBools.add(true);
+            toLogCommandStatus.put(command, toLogBools);
+        });
+        CommandScheduler.getInstance().onCommandFinish(command -> {
+            Queue<Boolean> toLogBools = toLogCommandStatus.getOrDefault(command, new LinkedList<>());
+            toLogBools.add(false);
+            toLogCommandStatus.put(command, toLogBools);
+        });
+        CommandScheduler.getInstance().onCommandInterrupt(command -> {
+            Queue<Boolean> toLogBools = toLogCommandStatus.getOrDefault(command, new LinkedList<>());
+            toLogBools.add(false);
+            toLogCommandStatus.put(command, toLogBools);
+        });
+        hasInit = true;
+    }
+
+    public void log() {
+        Iterator<Map.Entry<Command, Queue<Boolean>>> iterator = toLogCommandStatus.entrySet().iterator();
+        while (iterator.hasNext()){
+            Map.Entry<Command, Queue<Boolean>> entry = iterator.next();
+            Boolean poll = entry.getValue().poll();
```

Comment:
Why only look at the head of the queue? If we're already here and we have a queue shouldn't we get all values?

Reply:
This was a solution to the following problem: If a command starts and ends on the same tick, it would only log "init” or "end" state. This is because the boolean log value can only have one value per tick. To remedy this, I wait a tick to record the End state.